pymongo/client_session.py

```diff
@@ -1097,24 +1110,25 @@ def pop_all(self) -> list[_ServerSession]:
             ids.append(self.pop().session_id)
         return ids
 
-    def get_server_session(self, session_timeout_minutes: float) -> _ServerSession:
+    def get_server_session(self, session_timeout_minutes: Optional[int]) -> _ServerSession:
         # Although the Driver Sessions Spec says we only clear stale sessions
         # in return_server_session, PyMongo can't take a lock when returning
         # sessions from a __del__ method (like in Cursor.__die), so it can't
         # clear stale sessions there. In case many sessions were returned via
         # __del__, check for stale sessions here too.
-        self._clear_stale(session_timeout_minutes)
+        if session_timeout_minutes is not None:
+            self._clear_stale(session_timeout_minutes)
 
-        # The most recently used sessions are on the left.
-        while self:
-            s = self.popleft()
-            if not s.timed_out(session_timeout_minutes):
-                return s
+            # The most recently used sessions are on the left.
+            while self:
+                s = self.popleft()
+                if not s.timed_out(session_timeout_minutes):
+                    return s
```

Comment:
We still want to acquire a session from the pool here even if the timeout is None.